client/jetpack-connect/signup.js

```
@@ -76,31 +77,35 @@ export class JetpackSignup extends Component {
 
 	handleSubmitSignup = ( form, userData ) => {
 		debug( 'submiting new account', form, userData );
-		this.props.createAccount( userData );
+		this.props.createAccount( userData ).then(
+			bearerToken =>
+				this.setState( {
+					newUsername: userData.username,
+					bearerToken,
+				} ),
+			error => {
+				debug( 'Signup error: %o', error );
+				this.props.errorNotice( 'Danger!!!' );
```

Do something better. Errors were not handled before 😱